microbench/src/main/java/io/netty/microbench/buffer/ByteBufUtilBenchmark.java

```diff
@@ -164,13 +166,25 @@ public void writeUtf8ViaArrayWrapped() {
     @Benchmark
     public void writeUtf8() {
         buffer.resetWriterIndex();
-        ByteBufUtil.writeUtf8(buffer, utf8Sequence);
+        ByteBufUtil.writeUtf8(buffer, utf8);
+    }
+
+    @Benchmark
+    public void writeAsciiAsUtf8() {
```

Comment:
Those benchmarks are wrong. More info why - http://hg.openjdk.java.net/code-tools/jmh/file/fbe1b55eadf8/jmh-samples/src/main/java/org/openjdk/jmh/samples/JMHSample_08_DeadCode.java

Reply:
I agree on the paper and I was thinking to push another PR following the JMH best practices.
TBH what is happening in most of the already provided ones is different, thanks to `@TearDown` that for the JVM to avoid the benchmarked methods to be removed: the JVM cannot declare them as *dead code* because it can't say in advance what is happening on the tear down method.
This is confirmed by looking at the assembly produced, that shows on C2 optimization level that the code is not removed.

Reply:
Re http://hg.openjdk.java.net/code-tools/jmh/file/fbe1b55eadf8/jmh-samples/src/main/java/org/openjdk/jmh/samples/JMHSample_08_DeadCode.java it is a different case because the methods are side effect free from the calling method point of view

Reply:
Anyway considering that the results doesn't show big margins as I was expecting (in addition, I've a branch with changed JMH benchmarks too) I can close this PR :+1:

Reply:
@franz1981 are you sure it's within the error margin and not a 1000000 ops/s increase?

```code
ByteBufUtilBenchmark.writeAsciiAsUtf8  thrpt   20  12456281.779 ± 279760.847  ops/s
                                                     279760
ByteBufUtilBenchmark.writeAsciiAsUtf8  thrpt   20  13828353.123 ± 131895.610  ops/s
                                                     131895
```

Reply:
@johnou You're right, probably marking it as "within the error margin" is too harsh :P
My concern is that it is not a *huge* improvement as I was expecting, just ~10%...

Reply:
For us that's huge, ByteBufUtil.writeUtf8 is a hot path. Could I anyway request that you change the jmh to return the object to avoid copy / paste and running into that dead code trap?

Reply:
Hard to see from mobile but does the improvement come from the fact that writeUtf8 can now be inlined?

Reply:
Mmm difficult to say...The original bytecode size was small enough that it was already inlined: if I've to put my money I will bet it is due to a better usage of the micro ops cache of the CPU...
I've to run th Perf profiler to be sure of that, hoping that my consumer CPU has that kind of counters available!